icicle/src/msm/msm.cu

```diff
@@ -846,32 +887,25 @@ namespace msm {
   }
 
   template <typename A, typename P>
-  cudaError_t precompute_msm_bases(
-    A* bases,
-    int bases_size,
-    int precompute_factor,
-    int _c,
-    bool are_bases_on_device,
-    device_context::DeviceContext& ctx,
-    A* output_bases)
+  cudaError_t precompute_msm_bases(A* bases, int msm_size, MSMConfig& config, A* output_bases)
```

Comment:
After discussing with the team, in order to prevent a breaking change right now, we think its best to:
1. create a second function that will call the old function after computing `c`
2. add a deprecation comment on the old function

Reply:
actually, why do we need to call the old function? I will just give my new precompute function a different name and also keep the old one as deprecated.

Reply:
just so you don't implement twice.